jujugui/static/gui/src/app/app.js

```diff
@@ -2651,18 +2643,15 @@ YUI.add('juju-gui', function(Y) {
       // authenticated. If we aren't then display the login screen.
       const shouldDisplayLogin = apis.some(api => {
         // Legacy Juju won't have a controller API.
-        if (!api) {
+        // If we do not have an api instance or if we are not connected with
+        // it then we don't need to concern ourselves with being
+        // authenticated to it.
+        if (!api || !api.get('connected')) {
           return false;
         }
         // If the api is connecting then we can't know if they are properly
         // logged in yet.
         if (api.get('connecting')) {
-          return true;
-        }
-        if (!api || !api.get('connected')) {
-          // If we do not have an api instance or if we are not connected with
-          // it then we don't need to concern ourselves with being
-          // authenticated to it.
           return false;
```

Comment:
If we're still connecting then do not show the login screen as we're often logged into the controller but not the model causing the app to dispatch incorrectly. It's almost impossible to be logged into a controller but then not be able to log into a model you have access to as per the controller.